guides/hack/05-functions/02-anonymous-functions.md

```diff
@@ -1,9 +1,40 @@
 Hack supports anonymous functions.
 
+In the example below, the anonymous function, `$f`, evaluates to a function that
+returns the value of `$x + 1`.
+
 ``` Hack
 $f = $x ==> $x + 1;
 
-$two = $f(1);
+$two = $f(1); // result of 2
+```
+
+## Default Functionality: Pass by Value
+Anonymous functions pass _by value_, not by reference, as shown in the example below.
```

Comment:
I know @jjergus used the term "pass by value" in #879, but I think it's clearer to talk about scope. I chatted about this with a few other Hacklang people and I'm not sure we have a standard term for this.

```
$v = vec[];
$f = () ==> { $v[0] = 1; }
$f(); // $v is vec[]

$v = Vector {};
$f = () ==> { $v[0] = 1; }
$f(); // $v is Vector { 1 }
```

Value versus reference semantics depends on the type of the value you're closing over. Similarly an object has reference semantics:

```
$o = new MyObject(5);
$f = () ==> { $o->value = 6; }
$f();

$o->value; // 5
```

The important thing here is that captured variables take the value when the closure is evaluated. This is extremely unusual: only PHP and Hack do this in a way that's observable to user code.

Reply:
This is still a bit confusing to me. Just speaking out loud, likely wrong understanding, the above example has `vec`, an array, and `Vector`, a collection. We're saying that Hack has pass by value-like behavior (doesn't persist) for `vec[]` and pass by reference-like behavior (persists) for `Vector{}`. 

Then, later, the next example, we have an object, which should also be similar to collection behavior because both are object types, right?  But the reference doesn't persist... because for closures we need to see where the variable is assigned, not called. My head is all sorts of jumbled here.

Reply:
```
$v = vec[];
$f = () ==> { $v[] = 1; }
$v[] = 2;

$f(); // $v is vec[2]
```